packages/editor/src/components/rich-text/index.js

```diff
@@ -367,15 +367,20 @@ export class RichText extends Component {
 		}
 
 		let { selectedFormat } = this.state;
-		const { formats, text, start, end } = this.patterns.reduce(
+
+		const { formats, lineFormats, objects, text, start, end } = this.patterns.reduce(
 			( accumlator, transform ) => transform( accumlator ),
 			this.createRecord()
 		);
 
 		if ( this.formatPlaceholder ) {
-			formats[ this.state.start ] = formats[ this.state.start ] || [];
-			formats[ this.state.start ].push( this.formatPlaceholder );
-			selectedFormat = formats[ this.state.start ].length;
+			selectedFormat = this.formatPlaceholder.length;
```

Comment:
From context, `selectedFormat` seems to be used as a boolean. If so:
- it should be renamed as `isSelectedFormat` or `isFormatSelected`, depending on intended meaning
- the RHS of the assignment should cast to boolean, e.g. `!! this.formatPlaceholder.length` or `length > 0`

Reply:
`selectedFormat` is should be a natural number. How can I make this clearer? I could change the checks for it.

Reply:
I don't know the stack well enough to advise :)